documentation/modules/ROOT/pages/transformations/event-changes.adoc

```diff
@@ -0,0 +1,147 @@
+** Timestamp that identifies when the change was made.
+** Optional transaction information.
+
+* Row data before the change
+* Row data after the change
+
+The following example shows part of the structure of a typical {prodname} `UPDATE` change event:
+
+[source,json,indent=0]
+----
+{
+	"op": "u",
+	"source": {
+		...
+	},
+	"ts_ms" : "...",
+	"before" : {
+		"field1" : "oldvalue1",
+		"field2" : "oldvalue2"
+	},
+	"after" : {
+		"field1" : "newvalue1",
+		"field2" : "newvalue2"
+	}
+}
+----
+
+ifdef::community[]
+More details about change event structure are provided in
+xref:{link-connectors}[the documentation for each connector].
+endif::community[]
+
+The complex format of the message in the preceding example provides detailed information about changes that occur in the source database.
+However, the format might not be suitable for some downstream consumers.
+Sink connectors, or other parts of the Kafka ecosystem might expect the message to explicitly identify the fields that a database operation changes or leaves unchanged.
+The `ExtractChangedRecordState` SMT adds headers to the change event message to identify the fields that are modified by a database operation, and the fields that remain unchanged.
+
+[[event-changes-behavior]]
+== Behavior
+
+The event changes SMT extracts the `before` and `after` fields from a {prodname} `UPDATE` change event in a Kafka record.
+The SMT examines the fields in the `before` and `after` event state structures, and then writes the names of fields that changed or are unchanged based on the values of  settings that you apply to the SMT configuration.
+If the event represents an `INSERT` or `DELETE`, this single message transformation has no effect.
+
+You can configure the event changes SMT for a {prodname} connector, or for a sink connector that consumes messages emitted by a {prodname} connector.
+Configure the event changes SMT for a sink connector if you want Apache Kafka to retain the entire original {prodname} change events.
+The decision to apply the SMT to a source or sink connector depends on your particular use case.
+
+Depending on your use case, you can configure the transformation to modify the original message by performing one or both of the following tasks:
+
+* Identify the fields that are changed by an `UPDATE` event by listing them in the user-configured xref:extract-changes-header-changed-name[`header.changed.name`] header.
+* Identify the fields that are not changed by an `UPDATE` event by listing them in the user-configured xref:extract-changes-header-unchanged-name[`header.unchanged.name`] header.
+
+== Configuration
+
+Configure the {prodname} event changes SMT in a Kafka Connect source or sink connector by adding the SMT configuration details to your connector's configuration.
+To obtain the default behavior which doesn't add any headers, in a `.properties` file, you would specify something like the following:
```

Comment:
I don't think we should talk about `.properties` file. It is connector configuration.

Reply:
This is copied in from something that I introduced in some of the other transformation docs a couple of years ago. It appears in most of the other docs. We can remove it here and on L103 @Naros, and I'll open a Jira to correct instances that appear elsewhere.

Reply:
I believe these are just copied from the `ExtractNewRecordState` SMT documentation and amended to be relevant to this SMT, but the formatting is relatively the same.  I see Bob has provided some edits below, I'll adjust accordingly.